config.c

```diff
@@ -1587,7 +1587,7 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
 			if (newlog->oldDir) {
 				for (i = 0; i < newlog->numFiles; i++) {
 					char *ld;
-					dirName = ourDirName(newlog->files[i]);
+					dirName = dirname(strdup(newlog->files[i]));
```

Comment:
Good point about invalid free()'s. Commit 17ad612c92ea443ca49f62ab85156bcf068fdaaf should be more appropriate.